tools/hpcc_aks_util.pl

```diff
@@ -0,0 +1,772 @@
+Mem Req(Mi)     - Memory requested by this pod in MiB (affects scheduling)
+
+NODE DETAILS (when --show-nodes is used):
+========================================
+
+Node Name       - Kubernetes node name within the pool
+CPU Capacity    - Total CPU cores available on this specific node
+CPU Allocatable - CPU cores available for pod scheduling (after system overhead)
+CPU Used        - Actual CPU cores currently being consumed on this node
+CPU(%)          - Percentage of allocatable CPU being used on this node
+Mem Capacity(Mi)- Total memory available on this specific node (in MiB)
+Mem Alloc(Mi)   - Memory available for pod scheduling (after system overhead, in MiB)
+Mem Used(Mi)    - Actual memory currently being consumed on this node (in MiB)
+Mem(%)          - Percentage of allocatable memory being used on this node
+VM Size         - Azure VM instance type for this node
+
+For each node, ALL pods running on that node are listed (from all namespaces), showing:
+Namespace       - Kubernetes namespace the pod belongs to
+Pod Name        - Kubernetes pod name
+Phase           - Pod lifecycle phase (Running, Pending, Failed, etc.)
+Status          - Pod readiness status (Ready/NotReady)
+CPU Req         - CPU cores requested by this pod (affects scheduling)
+Mem Req(Mi)     - Memory requested by this pod in MiB (affects scheduling)
+
+COST CALCULATIONS:
+=================
+
+The script uses East US 2024 pricing as baseline and applies configurable discounts:
+
+--ea-discount           Applied to most VM series (D-series v5, v4 smaller, E-series, F-series, B-series)
+--ds_v4_family_discount Applied specifically to D16ds v4 and larger instances
+
+Default discounts are 10% each. Costs are estimates and actual pricing may vary.
+
+MESSAGES SECTION:
+================
+
+The script provides actionable insights:
+
+- Resource Request Issues: Identifies pods without CPU requests (prevents proper scheduling)
+- Pod Health: Reports pods that are not ready with reasons
+- Thor Conflicts: Detects thor agents and workers/managers in same pool (prevents scaling to 0)
+- Resource Waste: Identifies pools with no target namespace workloads but consuming CPU
+
+UNDERSTANDING THE METRICS:
+=========================
+
+Key efficiency indicators:
+- HPCC(%) shows how much of the pool is used by your target namespace
+- InUse(%) shows actual CPU consumption vs. what's available
+- High Alloc(%) with low InUse(%) may indicate over-provisioning
+- Cost/Hr helps identify expensive pools that might be candidates for optimization
+
+For optimal resource utilization:
+- HPCC(%) should be high for dedicated pools
+- Pods should have appropriate CPU/memory requests
+- No thor conflicts should exist
+- Consider scaling down pools with 0% HPCC but >0% InUse
+
+EOF
+}
+
+# Initialize variables
+my $resource_group = "";
+my $cluster_name = "";
+my $namespace = "";
+my $show_pods = 0;
+my $show_nodes = 0;
+my $ea_discount = 10;
+my $ds_v4_family_discount = 10;
```

Comment:
see prev. comment - wouldn't it be better not to apply default [wrong] discounts, i.e. better for these values to be 0?